src/coaster.cpp

```diff
@@ -896,3 +937,68 @@ int CoasterInstance::GetNumberOfCars() const
 	if (number_cars == 0) number_cars = 1;
 	return number_cars;
 }
+
+void CoasterInstance::Load(Loader &ldr)
+{
+	this->RideInstance::Load(ldr);
+
+	this->capacity = (int)ldr.GetLong();
+	this->coaster_length = ldr.GetLong();
+
+	const CoasterType *ct = this->GetCoasterType();
+
+	uint saved_pieces = ldr.GetWord();
+	for (uint i = 0; i < saved_pieces; i++) {
+		uint32 index = ldr.GetLong();
+		this->pieces[i].piece = ct->pieces[index];
+		this->pieces[i].Load(ldr);
+		this->PlaceTrackPieceInWorld(this->pieces[i]);
+	}
+
+	int number_of_trains = (int)ldr.GetLong();
+	if (number_of_trains > 0) this->SetNumberOfTrains(number_of_trains);
```

Comment:
Maybe should fail in case of bad data, rather than just silently failing? Not sure if there's a facility for failing though...